### PR TITLE
Adding learning/riemannian_robust_m_estimator.py and testing cases, tutorial notebook.

### PR DESCRIPTION
**Pull request by jihyunnryu**
## Checklist

- [ o ] My pull request has a clear and explanatory title.
- [ o ] If necessary, my code is [vectorized](https://www.geeksforgeeks.org/vectorization-in-python/).
- [ o ] I added appropriate unit tests.
- [ o ] I made sure the code passes all unit tests. (refer to comment below)
- [ o ] My PR follows [PEP8](https://peps.python.org/pep-0008/) guidelines. (refer to comment below)
- [ o ] My PR follows [geomstats coding style](https://github.com/geomstats/geomstats/blob/main/docs/contributing.rst#coding-style-guidelines) and API.
- [ o ] My code is properly documented and I made sure the documentation renders properly. ([Link](https://github.com/geomstats/geomstats/blob/main/docs/contributing.rst#documentation))
- [ ] I linked to issues and PRs that are relevant to this PR.

## Description
Computing robust M-estimators on Riemannian manifolds.

Even on Riemannian manifolds, dataset can be contaminated easily in the real world scenario. However, there are few computing methods for center of mass or variance considering outliers. Thus, this code can be utilized for ones who desire to get statistics on manifolds with controlled outlier impact.

In the code, there are several highly recommended and commonly used M-estimator functions like Huber loss, Tukey's biweight loss and gradient functions respectively so that we can estimate M-estimator parameters in a few seconds.

Moreover, when the function is given properly with required input parameters, anyone can automatically compute M-estimator with auto gradient descent algorithms to give more degree of freedom by added autograd version of Riemannian Gradient descent algorithm .
So anyone can utilize gradient descent algorithm with desired M-estimators or loss functions to test to get the center of mass.

Testing cases and a tutorial notebook is in this PR, and I tested and found all passed.

### CODE REVIEW

**Comment by jihyunnryu** [comment]
I'm wondering if it's taking a long time because the code didn't meet your criteria. I'd like to know the standards you use to review code and approve a PR. If I knew them, I could modify the code to meet those standards, but right now I can't do anything because I don't know what's preventing it from moving forward. I can't find detailed standards in the contributing guide in the documentation. So far on my environment, testing code works well but I can't anticipate the coverage that would be given by codecov.

File: geomstats/learning/riemannian_robust_m_estimator.py [hunk 1]
```diff
@@ -0,0 +1,1568 @@
...
+                )
+
+            continuing_condition = gs.less_equal(self.epsilon * space.dim, tangent_norm)
+            if not (continuing_condition or current_iter == 0):
+                break
+
+            estimate_next = space.metric.exp(-1 * step_size * gradient_value, mean)
+
+            var = gs.sum(space.metric.squared_dist(points, mean)) / (n_points - 1)
+            sq_dist = space.metric.squared_norm(estimate_next, mean)
```
Comment (luisfpereira) [inline geomstats/learning/riemannian_robust_m_estimator.py]: Not used

File: geomstats/learning/riemannian_robust_m_estimator.py [hunk 2]
```diff
@@ -0,0 +1,1568 @@
...
+    c = 10**decimal
+    return gs.floor(array * c + 0.5) / c
+
+
+def _set_midpoint(points):
+    """Generate midpoint as initial point for Riemannian gradient descent.
+
+    Parameters
+    ----------
+    points : Given dataset.
```
Comment (luisfpereira) [inline geomstats/learning/riemannian_robust_m_estimator.py]: Check docstrings in other files in `geomstats`.
Please make them consistent.
This applies to most docstrings in this file.

File: geomstats/learning/riemannian_robust_m_estimator.py [hunk 3]
```diff
@@ -0,0 +1,1568 @@
+"""Riemannian Robust M-estimator Fitting.
```
Comment (luisfpereira) [inline geomstats/learning/riemannian_robust_m_estimator.py]: Make sure linting passes

**Review by luisfpereira — CHANGES_REQUESTED** [review]
Thank you very much for this huge contribution @jihyunnryu.

In this review I request some changes to the code structure that I believe will make it more `geomstats`-like.

I suggest you implement some of my comments and ask me to further review afterwards.

Feel free to ask me any question along the way.

File: geomstats/learning/riemannian_robust_m_estimator.py (hunk 3 above)
Comment (jihyunnryu) [reply geomstats/learning/riemannian_robust_m_estimator.py]: I revised the code as you guided. please review the updated code

**Comment by jihyunnryu** [comment]
@luisfpereira Is there any progress here? I have been waiting for 3 months and am ready to make additional corrections.
